src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx

```diff
@@ -0,0 +1,865 @@
+
+```bash showLineNumbers={false}
+
+npm install @aws-amplify/data-construct
+
+```
+
+**Step 5:** Update the `cdk_backend/lib/cdk-backend-stack.ts` file as shown in the following code to use the `AmplifyData` construct  to create an AWS AppSync API. 
+
+```bash title="lib/cdk-backend-stack.ts"
+
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { AmplifyData, AmplifyDataDefinition } from '@aws-amplify/data-construct';
+
+export class CdkBackendStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+ // highlight-start
+ new AmplifyData(this, 'AmplifyCdkData', {
+      definition: AmplifyDataDefinition.fromString(/* GraphQL */ `
+      type Todo @model @auth(rules: [{ allow: public }]) {
+          id: ID!
+          name: String!
+          description: String
+          complete: Boolean
+        }
+      `),
+      authorizationModes: {
+        defaultAuthorizationMode: 'API_KEY',
+        apiKeyConfig: {
+          expires: cdk.Duration.days(30)
+        }
+      }
+    });
+ // highlight-end
+  }
+}
+
+```
+
+**Step 6:** Deploy the CDK stacks by running the following command.
+
+```bash showLineNumbers={false}
+
+cdk deploy
+
+```
+
+**Step 7:** The CDK will prepare the resources for deployment and will display the following prompt. Enter **Y** and press **Enter**.
+
+<img src="/images/existing-resources/resources-cdk.png" alt="The CDK preparing for deployment." width="750" />
+
+The CDK will deploy the stacks and display the following confirmation. Note the details of the deployed API; we’re going to use them in the next section.
+
+<img src="/images/existing-resources/cdk-stacks.png" alt="CDK deploying the stacks." width="750" />
+
+Now that you have built the backend API with the CDK, you can connect a frontend. Refer to section two to connect a Flutter app or section three to connect a React app, or complete both to connect frontends in each framework.
+
+## Build a Flutter app and connect to the GraphQL API
+
+In this section, we will connect a Flutter web app to the GraphQL API we created with the CDK. First, we will initialize a Flutter project, define models matching our schema, and use Amplify to integrate CRUD operations. Then we will add UI pages to manage to-do items with queries and mutations. Finally, we will configure Amplify with our backend details and run the app to demonstrate full functionality with our existing API.
+
+**Step 1:** Create a Flutter app by running the following command in your terminal.
+
+```dart showLineNumbers={false}
+
+flutter create flutter_todo_app --platforms=web   
+
+```
+
+**Step 2:** Open the newly created Flutter app by running the following commands in your terminal. 
+
+```dart
+
+cd flutter_todo_app 
+code . -r
+
+```
+
+**Step 3**: Update the `pubspec.yaml` file in the app’s root directory to add the required dependencies, as shown in the following code.
+
+{/* cSpell:disable */}
```

Comment:
is cspell tripping on `pubspec`? should we add this to the dictionary? https://github.com/aws-amplify/docs/blob/main/cspell.json